lib/TPP/ConvertTppToXsmm.cpp

```diff
@@ -342,65 +342,85 @@ struct ConvertTppIdentityOp : public OpRewritePattern<tpp::IdentityOp> {
+        return {ldi, bCast};
+      }
+
+      if (shapeInput[0] == 1 && shapeOutput[0] > 1) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_COL;
+        int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>());
+        return {ldi, bCast};
+      }
+
+      if (shapeInput[0] == shapeOutput[0] && shapeInput[1] == shapeOutput[1]) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::NONE;
+        int64_t ldi = *getLeadingDim(inputType.cast<MemRefType>());
+        return {ldi, bCast};
+      }
     }
-
-    if (shapeInput[0] == 1 && shapeOutput[0] > 1) {
-      xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_COL;
-      int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>());
-      return {ldi, bCast};
-    }
-
-    if (shapeInput[0] == shapeOutput[0] && shapeInput[1] == shapeOutput[1]) {
-      xsmm::UnaryFlags bCast = xsmm::UnaryFlags::NONE;
-      int64_t ldi = *getLeadingDim(inputType.cast<MemRefType>());
-      return {ldi, bCast};
+    // Handle 1d-case.
+    if (shapeInput.size() == 1) {
+      if (shapeInput[0] == 1 && shapeOutput[0] > 1) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_SCALAR;
+        int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>());
+        return {ldi, bCast};
+      }
+      if (shapeInput[0] == shapeOutput[0]) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::NONE;
+        int64_t ldi = *getLeadingDim(inputType.cast<MemRefType>());
+        return {ldi, bCast};
+      }
     }
     assert(false && "failed to get ldi and bCast for identity");
   }
 
   LogicalResult matchAndRewrite(tpp::IdentityOp identityOp,
                                 PatternRewriter &rewriter) const override {
     Location loc = identityOp.getLoc();
-    // no conversion if identity is a scalar operation.
     Type outputType = identityOp.getOutput().getType();
-    MemRefType outputMemRefType = outputType.dyn_cast<MemRefType>();
-    if (!outputMemRefType || outputMemRefType.getRank() != 2)
-      return rewriter.notifyMatchFailure(identityOp, "not a 2-D memref type");
+    assert(outputType.isa<MemRefType>() && "expect a memref type");
+    MemRefType outputMemRef = outputType.cast<MemRefType>();
+    assert((outputMemRef.getRank() == 1 || outputMemRef.getRank() == 2) &&
+           "expect memref with rank 1 or 2");
 
+    int64_t m = (outputMemRef.getRank() == 2) ? outputMemRef.getShape()[0] : 1;
+    int64_t n = (outputMemRef.getRank() == 2) ? outputMemRef.getShape()[1]
+                                              : outputMemRef.getShape()[0];
     int64_t outputOffset;
     SmallVector<int64_t> outputStrides;
-    if (failed(
-            getStridesAndOffset(outputMemRefType, outputStrides, outputOffset)))
+    if (failed(getStridesAndOffset(outputMemRef, outputStrides, outputOffset)))
       return rewriter.notifyMatchFailure(identityOp, "not a strided memref");
-    if (outputStrides.back() != 1)
+    if (outputStrides.back() != 1) {
```

Comment:
Don't add braces for one and not the other, it's harder to read.

Also, what does this error message means: "most minor stride is != 1"?